test/Microsoft.AspNetCore.Mvc.Abstractions.Test/ModelBinding/ModelStateDictionaryTest.cs

```diff
@@ -1007,6 +1114,21 @@ public void ModelStateDictionary_AddsCustomErrorMessage_WhenModelStateSet_WithNo
 
         [Fact]
         public void ModelStateDictionary_NoErrorMessage_ForUnrecognizedException()
+        {
+            // Arrange
+            var dictionary = new ModelStateDictionary();
+            dictionary.SetModelValue("key", new string[] { "some value" }, "some value");
+
+            // Act
+            dictionary.TryAddModelError("key", new InvalidOperationException());
+
+            // Assert
+            var error = Assert.Single(dictionary["key"].Errors);
+            Assert.Empty(error.ErrorMessage);
```

Comment:
Add another assertion: `error.Exception` will be the same as the `InvalidOperationException` passed in